docs/content/guides/dagster/managing-ml.mdx

```diff
@@ -0,0 +1,296 @@
+
+---
+
+## Machine learning operations (MLOps)
+
+You might have thought about your data sources, feature sets, and the best model for your use case. Inevitably, you start thinking about how to make this process sustainable and operational and deploy it to production. You want to make the machine learning pipeline self-sufficient and have confidence that the model you built is performing the way you expect. Thinking about machine learning operations, or MLOps, is the process of making your model maintainable and repeatable for a production use case.
+
+### Automating ML model maintenance
+
+Whether you have a large or small model, Dagster can help automate data refreshes and model training based on your business needs.
+
+We can use [auto-materializing assets](/concepts/assets/asset-auto-execution) to update a machine learning model when the upstream data is updated. This can be done by setting the `AutoMaterializePolicy` to `eager`, which means that our machine learning model asset will be refreshed anytime our data asset is updated.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=eager_materilization_start endbefore=eager_materilization_end
+from dagster import AutoMaterializePolicy, asset
+
+
+@asset
+def my_data():
+    ...
+
+
+@asset(
+    auto_materialize_policy=AutoMaterializePolicy.eager(
+        max_materializations_per_minute=60
+    )
+)
+def my_ml_model(my_data):
+    ...
+```
+
+Some machine learning models might more be cumbersome to retrain; it also might be less important to update them as soon as new data arrives. For this, we can use a lazy auto-materialization policy which can be used in two different ways. The first, by using it with a `freshness_policy` as shown below. In this case, `my_ml_model` will only be auto-materialized once a week.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=lazy_materlization_start endbefore=lazy_materlization_end
+from dagster import AutoMaterializePolicy, asset, FreshnessPolicy
+
+
+@asset
+def my_other_data():
+    ...
+
+
+@asset(
+    auto_materialize_policy=AutoMaterializePolicy.lazy(),
+    freshness_policy=FreshnessPolicy(maximum_lag_minutes=7 * 24 * 60),
+)
+def my_other_ml_model(my_other_data):
+    ...
+```
+
+This can be useful if you know that you want your machine learning model retrained at least once a week. While Dagster allows you to refresh a machine learning model as often as you like, best practice is to re-train as seldomly as possible. Model retraining can be costly to compute and having a minimal number of model versions can reduce the complexity of reproducing results at a later point in time.
+
+This can be useful if you want your machine learning model to be refreshed whenever the model is being used for a prediction. In this case, we are using the model once a week for `predictions`, ensuring that `my_ml_model` is retrained before it is used.
```

Comment:
Just noticed that this paragraph and the one before it (starts on line 75) both begin with `This can be useful`. I recommend reworking the second a little bit to vary your sentence structure. It reads like a continuation of the first paragraph, which I'm not sure is the intention.

Reply:
removed some extra verbiage